File: tests/algorithms/x/Java/maths/floor.java
public class Main {

    static long floor(double x) {
        long i = ((Number)(x)).intValue();
        if (x - ((Number)(i)).doubleValue() >= 0.0) {
            return i;
        }
        return i - 1;
    }

    static void test_floor() {
        double[] nums = ((double[])(new double[]{1.0, -1.0, 0.0, 0.0, 1.1, -1.1, 1.0, -1.0, 1000000000.0}));
        long[] expected_1 = ((long[])(new long[]{1, -1, 0, 0, 1, -2, 1, -1, 1000000000}));
        long idx_1 = 0;
        while (idx_1 < nums.length) {
            if (floor(nums[(int)(idx_1)]) != expected_1[(int)(idx_1)]) {
                throw new RuntimeException(String.valueOf("floor test failed"));
            }
            idx_1 = idx_1 + 1;
        }
    }

    static void main() {
        test_floor();
        System.out.println(_p(floor(-1.1)));
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            main();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        if (v instanceof Double || v instanceof Float) {
            double d = ((Number) v).doubleValue();
            if (d == Math.rint(d)) return String.valueOf((long) d);
            return String.valueOf(d);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/Kotlin/maths/floor.kt
fun panic(msg: String): Nothing { throw RuntimeException(msg) }

fun floor(x: Double): Int {
    var i: Int = ((x.toInt())).toInt()
    if ((x - ((i.toDouble()))) >= 0.0) {
        return i
    }
    return i - 1
}

fun test_floor(): Unit {
    var nums: MutableList<Double> = mutableListOf(1.0, 0.0 - 1.0, 0.0, 0.0, 1.1, 0.0 - 1.1, 1.0, 0.0 - 1.0, 1000000000.0)
    var expected: MutableList<Int> = mutableListOf(1, 0 - 1, 0, 0, 1, 0 - 2, 1, 0 - 1, 1000000000)
    var idx: Int = (0).toInt()
    while (idx < nums.size) {
        if (floor(nums[idx]!!) != expected[idx]!!) {
            panic("floor test failed")
        }
        idx = idx + 1
    }
}

fun user_main(): Unit {
    test_floor()
    println(floor(0.0 - 1.1).toString())
}

fun main() {
    user_main()
}
